src/common/datasource/src/compression.rs

```diff
@@ -73,6 +74,74 @@ impl CompressionType {
         !matches!(self, &Self::UNCOMPRESSED)
     }
 
+    pub async fn encode(&self, content: impl AsRef<[u8]>) -> io::Result<Vec<u8>> {
```

Comment:
Can we use a macro for these boilerplate `encode`/`decode` methods?

Reply:
done